### PR TITLE
Production deployment

### PR DESCRIPTION
**Pull request by DariuszMusielak**
# Description
Production deployment:
- tooltips
- modal descriptions
- circle specs fix
- security issue fix 

### Before merge checklist
- [ ] CircleCI is passing
- [ ] Proper labels are set

### CODE REVIEW

File: app/views/user_mailer/notify_unread_notifications.html.haml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+%p
+  = "You have #{@unread_notifications_count} unread #{@notification_text}."
```
Comment (elharion) [inline app/views/user_mailer/notify_unread_notifications.html.haml]: decorator

File: app/views/user_mailer/notify_unread_notifications.html.haml [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+  = link_to "Go to notifications page.", notifications_url
+
+%p{ style: 'display: inline-block; padding-top: 6px; border-top: 1px solid; font-size: 0.9em; margin-top: 10px !important;' }
+  We send this email first day of every month (only if) you have some unread notifications.
```
Comment (netguru-hound) [inline app/views/user_mailer/notify_unread_notifications.html.haml]: Line is too long. [91/80]

File: app/views/user_mailer/notify_unread_notifications.html.haml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+%p
+  = "You have #{@unread_notifications_count} unread #{@notification_text}."
+  = link_to "Go to notifications page.", notifications_url
+
+%p{ style: 'display: inline-block; padding-top: 6px; border-top: 1px solid; font-size: 0.9em; margin-top: 10px !important;' }
```
Comment (netguru-hound) [inline app/views/user_mailer/notify_unread_notifications.html.haml]: Line is too long. [125/80]

File: app/views/user_mailer/notify_unread_notifications.html.haml (hunk 1 above)
Comment (netguru-hound) [inline app/views/user_mailer/notify_unread_notifications.html.haml]: `= "..."` should be rewritten as `...`

File: app/assets/stylesheets/components/btn-modal.scss [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    line-height: 25px;
+    color: $color-white;
+    text-shadow: none;
+    border: none;
+    border-radius: 15px;
+    background-color: $color-light-blue;
+    box-shadow: 0 3px 9px $color-black-transparent;
+    z-index: 100;
+
+    &:active, &:hover {
```
Comment (netguru-hound) [inline app/assets/stylesheets/components/btn-modal.scss]: Each selector in a comma sequence should be on its own single line

File: app/assets/stylesheets/components/btn-modal.scss [hunk 5]
```diff
@@ -0,0 +1,19 @@
+.btn-info {
+  &--modal {
+    @include position(fixed, null 25px 30px null);
+    padding: 0;
+    font-size: 30px;
+    line-height: 25px;
+    color: $color-white;
+    text-shadow: none;
+    border: none;
```
Comment (netguru-hound) [inline app/assets/stylesheets/components/btn-modal.scss]: `border: 0` is preferred over `border: none`

File: app/assets/stylesheets/components/btn-modal.scss [hunk 6]
```diff
@@ -0,0 +1,19 @@
+.btn-info {
+  &--modal {
+    @include position(fixed, null 25px 30px null);
+    padding: 0;
```
Comment (netguru-hound) [inline app/assets/stylesheets/components/btn-modal.scss]: Properties should be ordered background-color, border, border-radius, box-shadow, color, font-size, line-height, padding, text-shadow, z-index

File: spec/spec_helper.rb [hunk 7]
```diff
@@ -61,6 +61,11 @@
   config.enable_processing = false
 end
 
+# we have problems with geckodriver and newest version of firefox(52) new configuration is added to circle.yml in dependencies: pre: state
```
Comment (netguru-hound) [inline spec/spec_helper.rb]: Line is too long. [138/100]

File: lib/tasks/mailer.rake [hunk 8]
```diff
@@ -30,6 +30,23 @@ namespace :mailer do
     end
   end
 
+  desc 'Email about users that have some unread notifications'
+  task users_with_unread_notifications: :environment do
+    unread_notifications_count = %((select count(*) FROM notifications WHERE notifications.receiver_id = users.id AND notifications.notification_status = 'unread'))
```
Comment (netguru-hound) [inline lib/tasks/mailer.rake]: Line is too long. [164/100]

File: app/views/skills/index.html.haml [hunk 9]
```diff
@@ -29,20 +30,24 @@
...
+                        = link_to edit_draft_skill_path(skill.requested_change),
+                          class: "btn btn-danger #{update_request_btn_class(skill.requested_change, current_user)}",
+                          role: 'button' do
+                          %i.glyphicon.glyphicon-comment
                     - else
-                      = link_to edit_skill_path(skill),
-                                class: "btn btn-primary",
-                                role: 'button' do
-                        %i.glyphicon.glyphicon-cog
+                      .tooltip-container{ data: { toggle: 'tooltip', placement: 'top' }, title: 'Click to edit skill' }
```
Comment (netguru-hound) [inline app/views/skills/index.html.haml]: Line is too long. [139/80]<br>Line contains trailing whitespace

File: app/views/skills/index.html.haml [hunk 10]
```diff
@@ -29,20 +30,24 @@
...
-                                role: 'button' do
-                        %i.glyphicon.glyphicon-comment
+                      .tooltip-container{ data: { toggle: 'tooltip', placement: 'top' }, title: 'Access denied because there already is request for change' }
+                        = link_to edit_skill_path(skill),
+                          class: 'btn btn-primary disabled',
+                          role: 'button' do
+                          %i.glyphicon.glyphicon-cog
+                      .tooltip-container{ data: { toggle: 'tooltip', placement: 'top' }, title: 'Review request details for change' }                          
+                        = link_to edit_draft_skill_path(skill.requested_change),
+                          class: "btn btn-danger #{update_request_btn_class(skill.requested_change, current_user)}",
```
Comment (netguru-hound) [inline app/views/skills/index.html.haml]: Line is too long. [116/80]

File: app/views/skills/index.html.haml [hunk 11]
```diff
@@ -29,20 +30,24 @@
...
-                      = link_to edit_draft_skill_path(skill.requested_change),
-                                class: "btn btn-danger #{update_request_btn_class(skill.requested_change, current_user)}",
-                                role: 'button' do
-                        %i.glyphicon.glyphicon-comment
+                      .tooltip-container{ data: { toggle: 'tooltip', placement: 'top' }, title: 'Access denied because there already is request for change' }
+                        = link_to edit_skill_path(skill),
+                          class: 'btn btn-primary disabled',
+                          role: 'button' do
+                          %i.glyphicon.glyphicon-cog
+                      .tooltip-container{ data: { toggle: 'tooltip', placement: 'top' }, title: 'Review request details for change' }
```
Comment (netguru-hound) [inline app/views/skills/index.html.haml]: Line is too long. [159/80]<br>Line contains trailing whitespace

File: app/views/skills/index.html.haml [hunk 12]
```diff
@@ -29,20 +30,24 @@
...
                     - if skill.requested_change
-                      = link_to edit_skill_path(skill),
-                                class: "btn btn-primary disabled",
-                                role: 'button' do
-                        %i.glyphicon.glyphicon-cog
-                      = link_to edit_draft_skill_path(skill.requested_change),
-                                class: "btn btn-danger #{update_request_btn_class(skill.requested_change, current_user)}",
-                                role: 'button' do
-                        %i.glyphicon.glyphicon-comment
+                      .tooltip-container{ data: { toggle: 'tooltip', placement: 'top' }, title: 'Access denied because there already is request for change' }
```
Comment (netguru-hound) [inline app/views/skills/index.html.haml]: Line is too long. [157/80]

File: app/views/skills/index.html.haml [hunk 13]
```diff
@@ -29,20 +30,24 @@
                 %td= skill.description
                 %td= skill.rate_type
                 %td
-                  .btn-group.skills__actions{ role: "group", aria: { label: "Actions" } }
-                    = link_to skill, class: "btn btn-info", role: 'button' do
-                      %i.glyphicon.glyphicon-search
+                  .skills__actions
+                    .tooltip-container{ data: { toggle: 'tooltip', placement: 'top' }, title: 'Display skill details' }
```
Comment (netguru-hound) [inline app/views/skills/index.html.haml]: Line is too long. [119/80]

File: app/views/skills/_last_5_draft_skills.html.haml [hunk 14]
```diff
@@ -23,11 +23,13 @@
...
+                = link_to draft_skill, class: "btn btn-info", role: 'button' do
+                  %i.glyphicon.glyphicon-search
               - unless draft_skill.resolved?
-                = link_to edit_draft_skill_path(draft_skill),
-                          class: "btn btn-primary",
-                          role: 'button' do
-                  %i.glyphicon.glyphicon-edit
+                .tooltip-container{ data: { toggle: 'tooltip', placement: 'top' }, title: 'Review request details for change' }
+                  = link_to edit_draft_skill_path(draft_skill),
+                            class: "btn btn-primary #{update_request_btn_class(draft_skill, current_user)}",
```
Comment (netguru-hound) [inline app/views/skills/_last_5_draft_skills.html.haml]: Line is too long. [108/80]

File: app/views/skills/_last_5_draft_skills.html.haml [hunk 15]
```diff
@@ -23,11 +23,13 @@
...
+            .skills__actions
+              .tooltip-container{ data: { toggle: 'tooltip', placement: 'top' }, title: 'Display request details for change' }
+                = link_to draft_skill, class: "btn btn-info", role: 'button' do
+                  %i.glyphicon.glyphicon-search
               - unless draft_skill.resolved?
-                = link_to edit_draft_skill_path(draft_skill),
-                          class: "btn btn-primary",
-                          role: 'button' do
-                  %i.glyphicon.glyphicon-edit
+                .tooltip-container{ data: { toggle: 'tooltip', placement: 'top' }, title: 'Review request details for change' }
```
Comment (netguru-hound) [inline app/views/skills/_last_5_draft_skills.html.haml]: Line is too long. [127/80]

File: app/views/skills/_last_5_draft_skills.html.haml [hunk 16]
```diff
@@ -23,11 +23,13 @@
             - else
               %span.text-danger not resolved
           %th
-            .btn-group.skills__actions{ role: "group", aria: { label: "Actions" } }
-              = link_to draft_skill, class: "btn btn-info", role: 'button' do
-                %i.glyphicon.glyphicon-search
+            .skills__actions
+              .tooltip-container{ data: { toggle: 'tooltip', placement: 'top' }, title: 'Display request details for change' }
```
Comment (netguru-hound) [inline app/views/skills/_last_5_draft_skills.html.haml]: Line is too long. [126/80]

File: app/views/shared/_requested_changes_modal.html.haml [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+            %b.text-info created
+            %span - when requested change is new, and ready to review.
+          %li
+            %b.text-success accepted
+            %span - when requested change is accepted.
+          %li
+            %b.text-danger declined
+            %span - when requested change is declined.
+      .modal-footer
+        %button.btn.btn-primary{ data: { dismiss: 'modal' }, type: 'button' } Got it!
```
Comment (netguru-hound) [inline app/views/shared/_requested_changes_modal.html.haml]: Line is too long. [85/80]

File: app/views/shared/_requested_changes_modal.html.haml [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+  %i.glyphicon.glyphicon-question-sign
+#requestedChangeModal.modal.fade{ aria: { labelledby: 'requestedChangeModalLabel' }, role: 'dialog', tabindex: '-1' }
+  .modal-dialog{ role: 'document' }
+    .modal-content
+      .modal-header
+        %button.close{ aria: { label: 'Close' }, data: { dismiss: 'modal' }, type: 'button' }
+          %span{ aria: { hidden: 'true' } } ×
+        %h4#categoryModalLabel.modal-title Requested changes
+      .modal-body
+        %span Table represents all requested changes. We have two types of requested change:
```
Comment (netguru-hound) [inline app/views/shared/_requested_changes_modal.html.haml]: Line is too long. [92/80]

File: app/views/shared/_requested_changes_modal.html.haml [hunk 19]
```diff
@@ -0,0 +1,29 @@
+.btn.btn-info.btn-info--modal.btn-xs{ data: { target: '#requestedChangeModal', toggle: 'modal' }, type: 'button' }
+  %i.glyphicon.glyphicon-question-sign
+#requestedChangeModal.modal.fade{ aria: { labelledby: 'requestedChangeModalLabel' }, role: 'dialog', tabindex: '-1' }
+  .modal-dialog{ role: 'document' }
+    .modal-content
+      .modal-header
+        %button.close{ aria: { label: 'Close' }, data: { dismiss: 'modal' }, type: 'button' }
+          %span{ aria: { hidden: 'true' } } ×
+        %h4#categoryModalLabel.modal-title Requested changes
```
Comment (netguru-hound) [inline app/views/shared/_requested_changes_modal.html.haml]: Classes should be listed before IDs (.modal-title should precede #categoryModalLabel)

File: app/views/shared/_requested_changes_modal.html.haml [hunk 20]
```diff
@@ -0,0 +1,29 @@
+.btn.btn-info.btn-info--modal.btn-xs{ data: { target: '#requestedChangeModal', toggle: 'modal' }, type: 'button' }
+  %i.glyphicon.glyphicon-question-sign
+#requestedChangeModal.modal.fade{ aria: { labelledby: 'requestedChangeModalLabel' }, role: 'dialog', tabindex: '-1' }
+  .modal-dialog{ role: 'document' }
+    .modal-content
+      .modal-header
+        %button.close{ aria: { label: 'Close' }, data: { dismiss: 'modal' }, type: 'button' }
```
Comment (netguru-hound) [inline app/views/shared/_requested_changes_modal.html.haml]: Line is too long. [93/80]

File: app/views/shared/_requested_changes_modal.html.haml [hunk 21]
```diff
@@ -0,0 +1,29 @@
+.btn.btn-info.btn-info--modal.btn-xs{ data: { target: '#requestedChangeModal', toggle: 'modal' }, type: 'button' }
+  %i.glyphicon.glyphicon-question-sign
+#requestedChangeModal.modal.fade{ aria: { labelledby: 'requestedChangeModalLabel' }, role: 'dialog', tabindex: '-1' }
```
Comment (netguru-hound) [inline app/views/shared/_requested_changes_modal.html.haml]: Line is too long. [117/80]<br>Classes should be listed before IDs (.modal should precede #requestedChangeModal)

File: app/views/shared/_requested_changes_modal.html.haml [hunk 22]
```diff
@@ -0,0 +1,29 @@
+.btn.btn-info.btn-info--modal.btn-xs{ data: { target: '#requestedChangeModal', toggle: 'modal' }, type: 'button' }
```
Comment (netguru-hound) [inline app/views/shared/_requested_changes_modal.html.haml]: Line is too long. [114/80]

File: app/views/shared/_notifications_modal.html.haml [hunk 23]
```diff
@@ -0,0 +1,20 @@
...
+        %span On this page you can see list of notifications from 30 days. We have two different types of notifications:
+        %ul
+          %li
+            %b.text-info Skill updated
+            %span - you receive it when skill you have rated have been edited.
+          %li
+            %b.text-success Skill created
+            %span - you receive it when new skill have been added.
+      .modal-footer
+        %button.btn.btn-primary{ data: { dismiss: 'modal' }, type: 'button' } Got it!
```
Comment (netguru-hound) [inline app/views/shared/_notifications_modal.html.haml]: Line is too long. [85/80]

File: app/views/shared/_notifications_modal.html.haml [hunk 24]
```diff
@@ -0,0 +1,20 @@
...
+  %i.glyphicon.glyphicon-question-sign
+#categoryModal.modal.fade{ aria: { labelledby: 'categoryModalLabel' }, role: 'dialog', tabindex: '-1' }
+  .modal-dialog{ role: 'document' }
+    .modal-content
+      .modal-header
+        %button.close{ aria: { label: 'Close' }, data: { dismiss: 'modal' }, type: 'button' }
+          %span{ aria: { hidden: 'true' } } ×
+        %h4#categoryModalLabel.modal-title Notifications
+      .modal-body
+        %span On this page you can see list of notifications from 30 days. We have two different types of notifications:
```
Comment (netguru-hound) [inline app/views/shared/_notifications_modal.html.haml]: Line is too long. [120/80]

File: app/views/shared/_notifications_modal.html.haml [hunk 25]
```diff
@@ -0,0 +1,20 @@
+.btn.btn-info.btn-info--modal.btn-xs{ data: { target: '#categoryModal', toggle: 'modal' }, type: 'button' }
+  %i.glyphicon.glyphicon-question-sign
+#categoryModal.modal.fade{ aria: { labelledby: 'categoryModalLabel' }, role: 'dialog', tabindex: '-1' }
+  .modal-dialog{ role: 'document' }
+    .modal-content
+      .modal-header
+        %button.close{ aria: { label: 'Close' }, data: { dismiss: 'modal' }, type: 'button' }
+          %span{ aria: { hidden: 'true' } } ×
+        %h4#categoryModalLabel.modal-title Notifications
```
Comment (netguru-hound) [inline app/views/shared/_notifications_modal.html.haml]: Classes should be listed before IDs (.modal-title should precede #categoryModalLabel)

File: app/views/shared/_notifications_modal.html.haml [hunk 26]
```diff
@@ -0,0 +1,20 @@
+.btn.btn-info.btn-info--modal.btn-xs{ data: { target: '#categoryModal', toggle: 'modal' }, type: 'button' }
+  %i.glyphicon.glyphicon-question-sign
+#categoryModal.modal.fade{ aria: { labelledby: 'categoryModalLabel' }, role: 'dialog', tabindex: '-1' }
+  .modal-dialog{ role: 'document' }
+    .modal-content
+      .modal-header
+        %button.close{ aria: { label: 'Close' }, data: { dismiss: 'modal' }, type: 'button' }
```
Comment (netguru-hound) [inline app/views/shared/_notifications_modal.html.haml]: Line is too long. [93/80]

File: app/views/shared/_notifications_modal.html.haml [hunk 27]
```diff
@@ -0,0 +1,20 @@
+.btn.btn-info.btn-info--modal.btn-xs{ data: { target: '#categoryModal', toggle: 'modal' }, type: 'button' }
+  %i.glyphicon.glyphicon-question-sign
+#categoryModal.modal.fade{ aria: { labelledby: 'categoryModalLabel' }, role: 'dialog', tabindex: '-1' }
```
Comment (netguru-hound) [inline app/views/shared/_notifications_modal.html.haml]: Line is too long. [103/80]<br>Classes should be listed before IDs (.modal should precede #categoryModal)

File: app/views/shared/_notifications_modal.html.haml [hunk 28]
```diff
@@ -0,0 +1,20 @@
+.btn.btn-info.btn-info--modal.btn-xs{ data: { target: '#categoryModal', toggle: 'modal' }, type: 'button' }
```
Comment (netguru-hound) [inline app/views/shared/_notifications_modal.html.haml]: Line is too long. [107/80]

File: app/views/shared/_categories_modal.html.haml [hunk 29]
```diff
@@ -0,0 +1,15 @@
...
+      .modal-header
+        %button.close{ aria: { label: 'Close' }, data: { dismiss: 'modal' }, type: 'button' }
+          %span{ aria: { hidden: 'true' } } ×
+        %h4#categoryModalLabel.modal-title Skill categories
+      .modal-body
+        %span Skill categories represents different sections at Netguru. Each section collets skills which we want to have as a company. We have one more special category.
+        %b Shared
+        %span which contains similar skills for couple categories.
+      .modal-footer
+        %button.btn.btn-primary{ data: { dismiss: 'modal' }, type: 'button' } Got it!
```
Comment (netguru-hound) [inline app/views/shared/_categories_modal.html.haml]: Line is too long. [85/80]

File: app/views/shared/_categories_modal.html.haml [hunk 30]
```diff
@@ -0,0 +1,15 @@
...
+  %i.glyphicon.glyphicon-question-sign
+#categoryModal.modal.fade{ aria: { labelledby: 'categoryModalLabel' }, role: 'dialog', tabindex: '-1' }
+  .modal-dialog{ role: 'document' }
+    .modal-content
+      .modal-header
+        %button.close{ aria: { label: 'Close' }, data: { dismiss: 'modal' }, type: 'button' }
+          %span{ aria: { hidden: 'true' } } ×
+        %h4#categoryModalLabel.modal-title Skill categories
+      .modal-body
+        %span Skill categories represents different sections at Netguru. Each section collets skills which we want to have as a company. We have one more special category.
```
Comment (netguru-hound) [inline app/views/shared/_categories_modal.html.haml]: Line is too long. [171/80]

File: app/views/shared/_categories_modal.html.haml [hunk 31]
```diff
@@ -0,0 +1,15 @@
+.btn.btn-info.btn-info--modal.btn-xs{ data: { target: '#categoryModal', toggle: 'modal' }, type: 'button' }
+  %i.glyphicon.glyphicon-question-sign
+#categoryModal.modal.fade{ aria: { labelledby: 'categoryModalLabel' }, role: 'dialog', tabindex: '-1' }
+  .modal-dialog{ role: 'document' }
+    .modal-content
+      .modal-header
+        %button.close{ aria: { label: 'Close' }, data: { dismiss: 'modal' }, type: 'button' }
+          %span{ aria: { hidden: 'true' } } ×
+        %h4#categoryModalLabel.modal-title Skill categories
```
Comment (netguru-hound) [inline app/views/shared/_categories_modal.html.haml]: Classes should be listed before IDs (.modal-title should precede #categoryModalLabel)

File: app/views/shared/_categories_modal.html.haml [hunk 32]
```diff
@@ -0,0 +1,15 @@
+.btn.btn-info.btn-info--modal.btn-xs{ data: { target: '#categoryModal', toggle: 'modal' }, type: 'button' }
+  %i.glyphicon.glyphicon-question-sign
+#categoryModal.modal.fade{ aria: { labelledby: 'categoryModalLabel' }, role: 'dialog', tabindex: '-1' }
+  .modal-dialog{ role: 'document' }
+    .modal-content
+      .modal-header
+        %button.close{ aria: { label: 'Close' }, data: { dismiss: 'modal' }, type: 'button' }
```
Comment (netguru-hound) [inline app/views/shared/_categories_modal.html.haml]: Line is too long. [93/80]

File: app/views/shared/_categories_modal.html.haml [hunk 33]
```diff
@@ -0,0 +1,15 @@
+.btn.btn-info.btn-info--modal.btn-xs{ data: { target: '#categoryModal', toggle: 'modal' }, type: 'button' }
+  %i.glyphicon.glyphicon-question-sign
+#categoryModal.modal.fade{ aria: { labelledby: 'categoryModalLabel' }, role: 'dialog', tabindex: '-1' }
```
Comment (netguru-hound) [inline app/views/shared/_categories_modal.html.haml]: Line is too long. [103/80]<br>Classes should be listed before IDs (.modal should precede #categoryModal)

File: app/views/shared/_categories_modal.html.haml [hunk 34]
```diff
@@ -0,0 +1,15 @@
+.btn.btn-info.btn-info--modal.btn-xs{ data: { target: '#categoryModal', toggle: 'modal' }, type: 'button' }
```
Comment (netguru-hound) [inline app/views/shared/_categories_modal.html.haml]: Line is too long. [107/80]

File: app/views/notifications/index.html.haml [hunk 35]
```diff
@@ -34,14 +35,14 @@
                   - else
                     %span.text-danger Not marked as read
                 %td
-                  .btn-group.skills__actions{ role: "group", aria: { label: "Actions" } }
+                  .tooltip-container{ data: { toggle: 'tooltip', placement: 'top' }, title: 'Display notification details' }
```
Comment (netguru-hound) [inline app/views/notifications/index.html.haml]: Line is too long. [124/80]

File: app/views/draft_skills/index.html.haml [hunk 36]
```diff
@@ -39,11 +40,13 @@
...
+                      = link_to draft_skill, class: "btn btn-info", role: 'button' do
+                        %i.glyphicon.glyphicon-search
                     - unless draft_skill.resolved?
-                      = link_to edit_draft_skill_path(draft_skill),
-                                class: "btn btn-primary #{update_request_btn_class(draft_skill, current_user)}",
-                                role: 'button' do
-                        %i.glyphicon.glyphicon-edit
+                      .tooltip-container{ data: { toggle: 'tooltip', placement: 'top' }, title: 'Show request details for change' }
+                        = link_to edit_draft_skill_path(draft_skill),
+                          class: "btn btn-primary #{update_request_btn_class(draft_skill, current_user)}",
```
Comment (netguru-hound) [inline app/views/draft_skills/index.html.haml]: Line is too long. [106/80]

**Pull request merged**